cvat-ui/src/utils/opencv-wrapper/intelligent-scissors.ts

```diff
@@ -0,0 +1,180 @@
+// Copyright (C) 2021 Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import { numberArrayToPoints, pointsToNumberArray, Point } from '../math';
+
+export interface IntelligentScissorsParams {
+    shape: {
+        shapeType: 'polygon' | 'polyline';
+    };
+    canvas: {
+        shapeType: 'points';
+        enableThreshold: boolean;
+        enableSliding: boolean;
+        allowRemoveOnlyLast: boolean;
+        minPosVertices: number;
+    };
+}
+
+export interface IntelligentScissors {
+    reset(): void;
+    run(points: number[], image: ImageData, offsetX: number, offsetY: number): number[];
+    params: IntelligentScissorsParams;
+}
+
+function applyOffset(points: Point[], offsetX: number, offsetY: number): Point[] {
+    return points.map(
+        (point: Point): Point => ({
+            x: point.x - offsetX,
+            y: point.y - offsetY,
+        }),
+    );
+}
+
+export default class IntelligentScissorsImplementation implements IntelligentScissors {
+    private cv: any;
+    private scissors: {
+        tool: any;
+        state: {
+            path: number[];
+            anchors: Record<
+            number,
+            {
+                point: Point;
+                start: number;
+            }
+            >; // point index : start index in path
+            image: any | null;
+        };
+    };
+
+    public constructor(cv: any) {
+        this.cv = cv;
+        this.scissors = {
+            // eslint-disable-next-line new-cap
+            tool: new cv.segmentation_IntelligentScissorsMB(),
+            state: {
+                path: [],
+                anchors: {},
+                image: null,
+            },
+        };
+
+        this.scissors.tool.setEdgeFeatureCannyParameters(32, 100);
+        this.scissors.tool.setGradientMagnitudeMaxLimit(200);
```

Comment:
Looks like `reset()` below doesn't reapply these settings

Reply:
Thx, fixed. Now it works better